notifications_utils/logging.py

```diff
@@ -172,6 +172,8 @@ class CustomLogFormatter(logging.Formatter):
     FORMAT_STRING_FIELDS_PATTERN = re.compile(r"\((.+?)\)", re.IGNORECASE)
 
     def add_fields(self, record):
+        if self._fmt is None:
```

Comment:
Was an exception raised before if self._fmt is None ? If not could this not break code that wouldn't be expecting this exception ? I'm not sure what this function does in all honesty but adding an exception might cause issues if its not expected

Reply:
MyPy complained here because `self._fmt` had type `Optional[str]` but the regex method `findall()` required a `str` for its first argument. 
I just tested this out and an exception would have been thrown here anyway if we call `findall(None)`:
![Screen Shot 2021-07-22 at 2 51 46 PM](https://user-images.githubusercontent.com/5498428/126693439-f0951b26-90d0-4478-8521-45f6f4234685.png)

Reply:
Okay awesome perhaps its better raised as a TypeError then such that if a try catch block is expecting TypeError it doesn't break

Reply:
Could we raise a TypeError here then ?

Reply:
Sure I can do that.